src/LineageHelper.ts

```diff
@@ -36,28 +47,27 @@ interface EdgeId {
 }
 
 /**
- * The element - table, column, edge - that is in focus
+ * A specific element in the lineage graph
  */
 export type ElementId = TableId | ColumnId | EdgeId;
 
-export type MatchedElement<TableData, ColumnData> = LineageElement<TableData, ColumnData> | ColumnReference<ColumnData>;
-
+/**
+ * Reference to a column
+ * Includes the id of the table it is part of
+ */
 export interface ColumnReference<ColumnData> {
   type: "column";
   tableId: string;
   column: Column<ColumnData>;
 }
 
-export const toElementId = <TableData, ColumnData>(element: MatchedElement<TableData, ColumnData>): ElementId => {
-  switch (element.type) {
-    case "table":
-      return { type: "table", tableId: element.id };
-    case "column":
-      return { type: "column", tableId: element.tableId, columnId: element.column.id };
-    case "edge":
-      return element;
-  }
-};
+type ConnectedElement<TableData, ColumnData> = Table<TableData, ColumnData> | ColumnReference<ColumnData> | Edge;
```

Comment:
Do we need to combine tables, columns and edges in single type here as opposite to the Lineage?

Reply:
We can probably pass several handlers to walker and keep it separate.

Reply:
@pavelgubin I renamed this type to `LineageGraphElement`. I don't think there is a simple way around it as the `findElement` function returns this type. Otherwise, I'll have to do a lot of type juggling and refactoring keeping everything separate. 

`findElement` is also exposed by the LineageHelper as a useful function for developers using rhombic.